src/server/minecraft-server.js

```diff
@@ -0,0 +1,1503 @@
+        let args = [];
+        let startChar, endChar;
+
+        if (required) {
+            startChar = '<';
+            endChar = '>';
+        }
+        if (optional) {
+            startChar = '[';
+            endChar = ']';
+        }
+
+        while (start !== -1) {
+            start = line.indexOf(startChar, end);
+            end = line.indexOf(endChar, start);
+
+            if (start !== -1) {
+                let option = line.substr(start + 1, end - start - 1);
+                let options = option.concat(line.substr(end + 1, line.indexOf(' ', end + 1) - end - 1));
+                if (options.indexOf('|') !== -1) {
+                    let args2 = options.split('|');
+                    while ( (arg = args2.shift()) !== undefined ) {
+                        args.push(arg);
+                    }
+                } else {
+                    args.push(options);
+                }
+                start = line.indexOf(startChar, end);
+            }
+        }
+
+        return args;
+    }
+
+    async getEula () {
+        await this.log('Getting MinecraftServer EULA acceptance state...');
+
+        let eula;
+        let eulaUrlLine;
+        let line;
+        let lineNumber;
+        let properties = this.properties;
+        let minecraftDirectory = properties.settings.minecraftDirectory;
+
+        await this.log('Reading MinecraftServer eula.txt...');
+        try {
+            eula = await fs.readFile(path.join(minecraftDirectory, 'eula.txt'), 'utf8');
+            eula = eula.split(os.EOL);
+            for (lineNumber = 0; lineNumber < eula.length; lineNumber++) {
+                if (eula[lineNumber]) {
+                    eulaUrlLine = eula[lineNumber].split('https://');
+                    if (eulaUrlLine.length == 2) {
+                        properties.eulaUrl = 'https://' + eulaUrlLine[1].substr(0, eulaUrlLine[1].indexOf(')'));
+                        await this.log('MinecraftServer EULA location: ' + properties.eulaUrl);
+                    }
+                    line = eula[lineNumber].split('=');
+                    if (line.length == 2) {
+                        properties.acceptedEula = !!JSON.parse(line[1]);
+                        await this.log(`MinecraftServer EULA accepted? ${properties.acceptedEula}`);
+                    }
+                }
+            }
+            properties.eulaFound = true;
+            properties.installed = true;
+            return Promise.resolve();
+        } catch (err) {
+            await this.log('Failed to read eula.txt.');
+            await this.log(err);
+            await this.log('Minecraft probably needs to be run once to stage new files.');
+            await this.log('Use the web interface to start the Minecraft server and accept the license agreement.');
+            properties.eulaFound = false;
+            properties.installed = false;
+            return Promise.reject(err);
+        }
+    }
+
+    async getMinecraftVersions () {
```

Comment:
This didn't seem to work correctly in the quick tests I ran on the web interface. I don't know if the problem is here, in the api or the interface itself